indico/modules/events/registration/controllers/display.py

```diff
@@ -125,10 +125,18 @@ class RHParticipantList(RHRegistrationFormDisplayBase):
     def _is_checkin_visible(reg):
         return reg.registration_form.publish_checkin_enabled and reg.checked_in
 
+    def _picture_url(self, reg):
+        if reg.get_personal_data_picture():
+            return url_for('event_registration.participant_picture', reg)
+        return ''
+
     def _merged_participant_list_table(self):
         def _process_registration(reg, column_names):
             personal_data = reg.get_personal_data()
-            columns = [{'text': personal_data.get(column_name, '')} for column_name in column_names]
+            columns = [{'text': personal_data.get(column_name, '')} for column_name in column_names
+                       if column_name != 'picture']
+            if 'picture' in column_names:
+                columns.append({'text': self._picture_url(reg), 'is_picture': True})
```

Comment:
You cannot append this at the end of the list, it needs to be at the right place according to `column_names`. Otherwise you get this mismatch between the headings and the data:

![image](https://github.com/indico/indico/assets/179599/bd06022c-43a8-4d0f-96ea-ae6952906d0c)

Reply:
Fixed.